content/guides/weird_characters.adoc

```diff
@@ -0,0 +1,846 @@
+through http://briancarper.net/blog/449/[a bit of hackery].
+
+If you se `#` *at the end* of a symbol, then it is used to automatically
+generate a new symbol. This is useful inside macros to keep macro specifics
+from leaking into the userspace. A regular `let` will fail in a macro definition
+
+[source,clojure]
+----
+user=> (defmacro m [] `(let [x 1] x))
+#'user/m
+user=> (m)
+CompilerException java.lang.RuntimeException: Can't let qualified name: user/x, compiling:(NO_SOURCE_PATH:1)
+----
+
+Instead you need to append `#` to the end of the variable name and let Clojure
+generate a unique symbol for it:
+
+[source, clojure]
+----
+user=> (defmacro m [] `(let [x# 1] x#))
+#'user/m
+user=> (m)
+1
+user=>
+----
+
+If we expand this macro, we can see the `gensym` 'd name:
+
+[source, clojure]
+----
+user=> (macroexpand '(m))
+(let* [x__681__auto__ 1] x__681__auto__)
+----
+
+Another place you'll see the `#` is in
+<<xref/../../reference/reader#tagged_literals,tagged literals>>
+Most commonly you'll see this use in https://github.com/edn-format/edn[EDN]
+(extensible data notation - a rich data fromat that can be used in Clojure)
+and in ClojureScript (`#js`). See  <<xref/../weird_characters#tagged_literals,`#inst`, `#uuid`, or `#js`>> for more info.
+
+* <<xref/../../reference/reader#tagged_literals,Clojure Documentation Reader>>
+* http://briancarper.net/blog/449/[Clojure Reader Macros]
+* http://clojuredocs.org/clojure_core/clojure.core/gensym[ClojureDocs - gensyms]
+
+== `#{` - Set macro
+
+See <<xref/../weird_characters#dispatch,(`#`)>> for additional details.
+
+`#{` defines a set (a collection of unique values) specifically a `hash-set`. The
+following are equivalent:
+
+[source, clojure]
+----
+user=> #{1 2 3 4}
+#{1 2 3 4}
+user=> (hash-set 1 2 3 4)
+#{1 2 3 4}
+----
+
+Attempting to create a `set` using this literal form will throw if there
+are duplicates. Instead the `hash-set` function should be used on a vector.
+
+[source, clojure]
+----
+user=> #{1 2 3 4 1}
+
+IllegalArgumentException Duplicate key: 1  clojure.lang.PersistentHashSet.createWithCheck (PersistentHashSet.java:68)
+user=> (set [1 2 3 4 1]) ; convert vector to set, removing duplicates
+#{1 2 3 4}
+----
+
+* <<xref/../../reference/data_structures#sets,Clojure Documentation: Sets>>
+
+== `#_` - Discard macro
+
+See <<xref/../weird_characters#dispatch,(`#`)>> for additional details.
+
+`#_` tells the reader to ignore the next form completely.
+
+[source,clojure]
+----
+user=> [1 2 3 #_ 4 5]
+[1 2 3 5]
+----
+Note that the space following `#_` is optional, so
+[source,clojure]
+----
+user=> [1 2 3 #_4 5]
+[1 2 3 5]
+----
+also works. Also note that the discard macro works for EDN
+
+The docs suggest that "The form following `#_` is completely skipped by the reader
+(This is a more complete removal than the `comments` macro which yields `nil`).".
+This can prove useful for debugging situations or for multiline comments.
```

Comment:
A useful trick to mention here is that multiple `#_` can be "stacked" to omit multiple forms: `{:a 1, #_#_ :b 2, :c 3}`